internal/build/docker.go

```diff
@@ -95,6 +99,42 @@ func (l *localDockerBuilder) buildBaseWithMounts(ctx context.Context, baseDocker
 	return getDigestFromOutput(output.String())
 }
 
+func (l *localDockerBuilder) buildImageWithSteps(ctx context.Context, baseDigest digest.Digest, tag string, steps []Cmd) (digest.Digest, error) {
+	imageWithSteps := baseDigest
+	for _, s := range steps {
+		resp, err := l.dcli.ContainerCreate(ctx, &container.Config{
+			Image: string(baseDigest),
+			Cmd:   s.argv,
+		}, nil, nil, "")
+		if err != nil {
+			return "", nil
+		}
+		containerID := resp.ID
+
+		err = l.dcli.ContainerStart(ctx, containerID, types.ContainerStartOptions{})
+		if err != nil {
+			return "", err
+		}
+
+		statusCh, errCh := l.dcli.ContainerWait(ctx, containerID, container.WaitConditionNotRunning)
+		select {
+		case err := <-errCh:
+			if err != nil {
+				return "", err
+			}
+		case <-statusCh:
+		}
```

Comment:
doesn't need to happen now, but i bet we can extract `createContainerWithCommand` into its own func?